plugin/src/main/java/org/opensearch/ml/settings/MLCommonsSettings.java

```diff
@@ -127,4 +127,22 @@ private MLCommonsSettings() {}
             Setting.Property.NodeScope,
             Setting.Property.Dynamic
         );
+
+    public static final Setting<List<String>> ML_COMMONS_REMOTE_MODEL_ELIGIBLE_NODE_ROLES = Setting
```

Comment:
Can we add some comments for these settings (summary of what you have in the PR description) so that it can't be confusing later? 

Will we deprecate `ML_COMMONS_ONLY_RUN_ON_ML_NODE` in the long run?

Reply:
Sure, will add some comment, deprecating that depends on if we have dedicated ML node.